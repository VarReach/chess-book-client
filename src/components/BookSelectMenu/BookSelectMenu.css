
#book-select__dropdown-scroll {
  height: 80%;
  width: 100%;
  margin-right: 5px;
  padding-right: 10px;
  overflow-y: auto;
  overflow-x: hidden;
}

#book-select__dropdown-scroll::-webkit-scrollbar {
  width: 6px;
  border-radius: 5px;
  background-color: #363e53;
}

#book-select__dropdown-scroll::-webkit-scrollbar-thumb {
  background-color: #c8c8c8;
  border-radius: 5px;
}

#book-select__dropdown-scroll::-webkit-scrollbar-thumb:hover {
  background-color: #9c9c9c;
}

.book-select__dropdown-menu button {
  width: 100%;
  border: none;
  font-family: 'Montserrat', sans-serif;
  background-color: transparent;
  outline: none;
  padding: 10px 15px;

  font-size: 14px;
  color: #f7f7f7;
  text-align: left;
  text-decoration: none;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}


.book-select__dropdown-menu button:hover {
  cursor: pointer;
  background-color: #4d5b75;
}

.book-select__dropdown-menu button:focus {
  background-color: #4d5b75;
}

.book-select__dropdown-menu button:disabled:hover, .book-select__dropdown-menu button:disabled {
  background-color: #2f343e;
}