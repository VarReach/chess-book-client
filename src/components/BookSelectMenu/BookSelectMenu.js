import React, { Component } from 'react';
import ChaptersContext from '../../contexts/ChaptersContext';
import './BookSelectMenu.css';

class BookSelectMenu extends Component {
  static contextType = ChaptersContext;

  renderBooks = () => {
    const books = Object.keys(this.context.books).map(bookId => {
      const book = this.context.books[bookId];
      const selectedBook = this.context.bookId;
      return (
        <li>
          <button key={bookId} disabled={bookId == selectedBook} onClick={() => this.handleOnClick(bookId)}>
            {bookId == selectedBook ? <i className="fas fa-book-open"/> : <i className="fas fa-book"/>}
            {book.title}
          </button>
        </li>)
    });
    return books;
  }

  handleOnClick = (bookId) => {
    this.context.setBookId(bookId);
  }

  render() {
    return (
      <div className="dropdown book-select__dropdown-menu">
        <div id="book-select__dropdown-scroll">
          <ul>
            {this.renderBooks()}
          </ul>
        </div>
      </div>
    )
  }
}

export default BookSelectMenu