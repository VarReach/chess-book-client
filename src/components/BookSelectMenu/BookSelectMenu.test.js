import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookSelectMenu from './BookSelectMenu';

describe('BookSelectMenu component', () => {
  it('renders the dropdown when state.showDropDown is true', () => {
    const wrapper = mount(
      <BookSelectMenu /> 
    );
    wrapper.find(BookSelectMenu).instance().showDropDown();
    wrapper.update();
    expect(toJson(wrapper.find('.book-select__dropdown-menu'))).toMatchSnapshot();
    wrapper.unmount();
  });
});