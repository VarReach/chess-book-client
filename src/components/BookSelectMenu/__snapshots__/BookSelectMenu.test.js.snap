// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSelectMenu component renders the dropdown when state.showDropDown is true 1`] = `
<div
  className="dropdown book-select__dropdown-menu"
>
  <div
    id="book-select__dropdown-scroll"
  >
    <ul />
  </div>
</div>
`;
